test/tests.spec.ts

```diff
@@ -3256,296 +3086,256 @@ describe("passport-saml /", function () {
-          }
-        });
-      });
-      it("errors if request has expired", function (done) {
-        this.clock.restore();
-        samlObj.validateRedirect(this.request, this.request.originalQuery, function (err) {
-          try {
-            should.exist(err);
-            err!.message!.should.eql("SAML assertion expired");
-            done();
-          } catch (err2) {
-            done(err2);
-          }
-        });
-      });
-      it("errors if request has a bad signature", function (done) {
-        this.request.Signature = "foo";
-        samlObj.validateRedirect(
-          this.request,
-          this.request.originalQuery,
-          function (err: Error | null) {
-            try {
-              should.exist(err);
-              err!.message!.should.eql("Invalid signature");
-              done();
-            } catch (err2) {
-              done(err2);
-            }
-          }
-        );
-      });
-      it("returns profile for valid signature including session index", function (done) {
-        samlObj.validateRedirect(this.request, this.request.originalQuery, function (err, profile) {
-          try {
-            should.not.exist(err);
-            profile!.should.eql({
-              ID: "_8f0effde308adfb6ae7f1e29b414957fc320f5636f",
-              issuer: "http://localhost:20000/saml2/idp/metadata.php",
-              nameID: "stavros@workable.com",
-              nameIDFormat: "urn:oasis:names:tc:SAML:1.1:nameid-format:emailAddress",
-              sessionIndex: "_00bf7b2d5d9d3c970217eecefb1194bef3362a618e",
-            });
-            done();
-          } catch (err2) {
-            done(err2);
-          }
-        });
+describe("validateRedirect()", function () {
+  describe("idp slo", function () {
+    let samlObj: SAML;
+    beforeEach(function () {
+      samlObj = new SAML({
+        cert: fs.readFileSync(__dirname + "/static/acme_tools_com.cert", "ascii"),
+        idpIssuer: "http://localhost:20000/saml2/idp/metadata.php",
       });
+      this.request = Object.assign(
+        {},
+        JSON.parse(fs.readFileSync(__dirname + "/static/idp_slo_redirect.json", "utf8"))
+      );
+      this.clock = sinon.useFakeTimers(Date.parse("2018-04-11T14:08:00Z"));
     });
-    describe("sp slo", function () {
-      let samlObj: SAML;
-      beforeEach(function () {
-        samlObj = new SAML({
-          cert: fs.readFileSync(__dirname + "/static/acme_tools_com.cert", "ascii"),
-          idpIssuer: "http://localhost:20000/saml2/idp/metadata.php",
-          validateInResponseTo: true,
-        });
-        this.request = Object.assign(
-          {},
-          JSON.parse(fs.readFileSync(__dirname + "/static/sp_slo_redirect.json", "utf8"))
+    afterEach(function () {
+      this.clock.restore();
+    });
+    it("errors if bad xml", async function () {
+      const body = {
+        SAMLRequest: "asdf",
+      };
+      try {
+        await samlObj.validateRedirectAsync(body, null);
```

Comment:
This test changed slightly. It used to include a valid redirect URL, but now just has `null`.